src/binder_data.c

```diff
@@ -2099,9 +2099,13 @@ binder_data_allow_submit_request(
             return TRUE;
         }
     } else if (allow) {
-        /* IRadioConfig >= 1.1 */
-        binder_data_request_queue
-            (binder_data_set_preferred_data_modem_new(data));
+        RADIO_INTERFACE_TYPE interface_type = radio_config_interface_type(data->dm->rc);
```

Comment:
wouldn't this be enough?
```diff
-    } else if (allow) {
+    } else if (allow && radio_config_interface_type(data->dm->rc) ==
+        RADIO_INTERFACE_TYPE_HIDL) {
```

Reply:
Actually the code it the pull request is not correct at the moment which I missed earlier and it will break AIDL support. It should be checking if type is RADIO_INTERFACE_TYPE_AIDL or type is RADIO_INTERFACE_TYPE_HIDL and interface >= 1.1.

Reply:
I've resolved quite a long if with separate method, please advise if that solution is acceptable to you or not